### PR TITLE
[metricbeat] Docker module meets metricbeat

### PR DESCRIPTION
**Pull request by douaejeouit**
Docker module wants to join you  :tada: !
Comming soon : Unit & integration tests for the rest of metricsets
@ruflin

### CODE REVIEW

File: metricbeat/docs/modules/system.asciidoc [hunk 1]
```diff
@@ -94,39 +94,39 @@ metricbeat.modules:
 
 The following metricsets are available:
 
-* <<metricbeat-metricset-system-core,core>>
+* <<metricbeat-metricset-system-diskio,diskio>>
```
Comment (ruflin) [inline metricbeat/docs/modules/system.asciidoc]: Not sure why these changes show up here as this should not have changed. I assume it is based on an old version of master.

File: metricbeat/module/docker/container/container.go [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+	mb.BaseMetricSet
+	dockerClient *dc.Client
+}
+
+// New create a new instance of the MetricSet
+// Part of new is also setting up the configuration by processing additional
+// configuration entries if needed.
+func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
+
+	config := docker.GetDefaultConf()
```
Comment (ruflin) [inline metricbeat/module/docker/container/container.go]: Lets mark is as experimental for now: https://github.com/elastic/beats/blob/master/metricbeat/module/beats/filebeat/filebeat.go#L26

File: metricbeat/module/docker/container/data.go [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+		myEvents = append(myEvents, eventMapping(&container))
+	}
+	return myEvents
+}
+func eventMapping(mycontainer *dc.APIContainers) common.MapStr {
+
+	event := common.MapStr{
+		"@timestamp": time.Now(),
+		"container": common.MapStr{
+			"container_created":    common.Time(time.Unix(mycontainer.Created, 0)),
```
Comment (ruflin) [inline metricbeat/module/docker/container/data.go]: container_ will have to be removed to follow our naming guidelines

File: metricbeat/module/docker/cpu/cpu.go [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+	mb.BaseMetricSet
+	cpuService   *CPUService
+	dockerClient *dc.Client
+}
+
+// New create a new instance of the MetricSet
+// Part of new is also setting up the configuration by processing additional
+// configuration entries if needed.
+func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
+	config := docker.GetDefaultConf()
```
Comment (ruflin) [inline metricbeat/module/docker/cpu/cpu.go]: Lets mark this experimental. For an example see: https://github.com/elastic/beats/blob/master/metricbeat/module/beats/filebeat/filebeat.go#L26

File: metricbeat/module/docker/diskio/diskio.go [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+	mb.BaseMetricSet
+	blkioService *BLkioService
+	dockerClient *dc.Client
+}
+
+// New create a new instance of the MetricSet
+// Part of new is also setting up the configuration by processing additional
+// configuration entries if needed.
+func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
+
```
Comment (ruflin) [inline metricbeat/module/docker/diskio/diskio.go]: Mark experimental, see comment above.

**Comment by ruflin** [comment]
To prevent future conflicts with master and review of large changes I suggest the following steps:

* Remove all parts from this PR which were accidentially added during merging like some config files
* Mark all metricsets experimental
* Check in source from docker client (not yet in the PR)
* Rebase on master / merge master

Then we can merge the PR as it is. There is still more work needed like some fields cleanup or adding tests. But that can be done in a second step and will make it much easier to review the PR's. Let me know in case someone has some objections?

@douaejeouit Could you take care of the 4 steps above?

File: metricbeat/module/docker/container/container.go (hunk 2 above)
Comment (douaejeouit) [inline metricbeat/module/docker/container/container.go]: +1

File: metricbeat/module/docker/container/data.go (hunk 3 above)
Comment (douaejeouit) [inline metricbeat/module/docker/container/data.go]: Ok :+1:

**Comment by douaejeouit** [comment]
@ruflin , Glad to read that! 
Concerning the tests, it's not fnished yet. I began with CPU & Memory metrisets and i'll  finish with the rest of metricsets.  
Ok for the 4 steps. i'll be on it as soon as possible :+1:

**Comment by douaejeouit** [comment]
poke @ruflin

**Comment by ruflin** [comment]
@douaejeouit It seems like the docker client has some more dependencies which are not checked in yet. That is the reason that the travis build fails.

**Comment by douaejeouit** [comment]
Here we are .. @ruflin

**Comment by ruflin** [comment]
jenkins, test it

**Comment by ruflin** [comment]
@douaejeouit I didn't check the PR in detail yet just had a look at the files change and it seems there are still quite a few files modified which shouldn't be modified by this PR. For example related to postgresql and others. This could come from merging in an older version of master or similar. Can you rebase your changes again on top of master and then run `make collect`? Please verify that no changes are in that should not be there. Because from my perspective I'm ok to merge in the docker module if it is not perfect yet, but it can't affect / change existing modules.

**Comment by douaejeouit** [comment]
@ruflin Ok. i'll do the needful

**Comment by ruflin** [comment]
@douaejeouit I took your branch here and run the suggested commands on top. See the outcome here: https://github.com/elastic/beats/pull/2619 Not sure if some of the commands on your side are not running successfully? Or did you not add them to git to be commited?

**Comment by ruflin** [comment]
@douaejeouit Thanks a lot for all your work on this one. We merge #2625 into master and I opened a meta issue to track the open issue / enhancements that should be made: https://github.com/elastic/beats/issues/2629 I hope now all our future PR's will be a little bit smaller :-) Closing this one.
